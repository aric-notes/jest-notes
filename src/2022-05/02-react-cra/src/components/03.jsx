export const Btn = ({ content, onClick, ...props }) => (
  <button {...props} onClick={onClick}>
    {content}
  </button>
);