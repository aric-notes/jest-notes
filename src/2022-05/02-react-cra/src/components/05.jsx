import styled from 'styled-components';

const StyledButton = styled.button`
  font-size: 1.5rem;
  border: none;
  padding: 1rem 2rem;
  border-radius: 4px;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
`;


export const Button = ({ children, ...props }) => {
  return <StyledButton {...props}>
    hi
  </StyledButton>;
};
