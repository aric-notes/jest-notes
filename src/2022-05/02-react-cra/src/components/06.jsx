import React from 'react';

// subtest: table
export const Button = ({ type }) => {
  const getCls = (type) => {
    const typeACls = 'eb-bg-green-500';
    switch (type) {
      case 'A':
        return typeACls;
      case 'B':
        return 'eb-bg-blue-500';
      case 'C':
        return 'eb-bg-red-500';
      default:
        return 'other';
    }
  };

  return (<button
    className={`${getCls(type)} eb-text-white eb-font-bold eb-py-2 eb-px-4 eb-rounded`}
  >
    Hi
  </button>);
};
