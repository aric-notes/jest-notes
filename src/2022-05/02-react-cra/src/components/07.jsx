import axios from 'axios';
import React, { useState, useEffect } from 'react';

export function getData() {
  return axios.get('http://my-backend/fake-data');
}

// onclick call back
export const Button = () => {
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);
  const [data, setData] = useState([]);

  return (
    <div>
      <button
        className='eb-bg-teal-500 hover:eb-bg-teal-700 eb-text-white eb-font-bold eb-py-2 eb-px-4 eb-rounded'
        onClick={() => {
          setIsLoading(true);
          getData()
            .then((response) => {
              setData(response.data);
            })
            .catch((e) => {
              setError(e.response.data);
            }).finally(() => {
            setIsLoading(false);
          });
        }}
      >
        請按
      </button>
      {isLoading ? (
        <div>Loading........</div>
      ) : (
        <ul>
          {data.map((i) => (
            <li className="item" key={i}>{i}</li>
          ))}
        </ul>
      )}
      {error && error}
    </div>
  );
};