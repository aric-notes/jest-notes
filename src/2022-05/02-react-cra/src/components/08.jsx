import { useState, useCallback } from 'react';

export function useCounter() {
  const [count, setCount] = useState(0);

  const increment = useCallback(() => setCount((x) => x + 1), []);

  return { count, increment };
}


const DemoCounter = () => {
  const { count, increment } = useCounter();

  return (
    <div>
      <div className='eb-text-black eb-text-purple-500 eb-font-bold eb-mr-4'>{count}</div>
      <button
        className='eb-bg-purple-500 hover:eb-bg-purple-700 eb-text-white eb-font-bold eb-py-2 eb-px-2 eb-rounded'
        onClick={increment}
      >
        increment
      </button>
    </div>
  );
};
export default DemoCounter;