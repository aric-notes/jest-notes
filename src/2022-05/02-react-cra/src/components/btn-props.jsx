export const BtnProps = ({ content, ...props }) => (
  <button {...props}>
    {content}
  </button>
);