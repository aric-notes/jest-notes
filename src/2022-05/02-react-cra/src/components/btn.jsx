export const Btn = ({ content, ...props }) => (
  <button {...props}>
    hi
  </button>
);