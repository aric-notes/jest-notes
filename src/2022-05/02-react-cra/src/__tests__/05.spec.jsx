import { render } from '@testing-library/react';
import { Button } from '../components/05';

test('btn should have hi text', () => {
  // 准备 Arrange

  // 执行 Act
  const { container } = render(<Button />);

  // 断言 Assert
  expect(container).toMatchSnapshot();
});