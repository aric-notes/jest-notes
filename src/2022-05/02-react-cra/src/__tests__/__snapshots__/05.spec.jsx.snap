// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`btn should have hi text 1`] = `
.c0 {
  font-size: 1.5rem;
  border: none;
  padding: 1rem 2rem;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

<div>
  <button
    class="c0"
  >
    hi
  </button>
</div>
`;
